### PR TITLE
Set member variable to Attention module

### PR DESCRIPTION
**Pull request by chunit-quic**
- Set and use self.n_heads from args.heads
- Use self.dim from args.dim

### CODE REVIEW

**Review by cccclai — APPROVED** [review]
Looks good to me. I'll trigger more test just in case.

**Review by helunwencser — APPROVED** [review]

File: examples/models/llama/llama_transformer.py [hunk 1]
```diff
@@ -260,21 +260,22 @@ class Attention(nn.Module):
...
+        self.head_dim = args.dim // self.n_heads
         self.max_batch_size = args.max_batch_size
         self.max_seq_len = args.max_seq_len
         self.dim = args.dim
-        # args.dim = 4096, args.n_heads = 32, self.head_dim = 4096 / 32 = 125
-        self.wq = nn.Linear(args.dim, args.n_heads * self.head_dim, bias=False)
-        self.wk = nn.Linear(args.dim, self.n_kv_heads * self.head_dim, bias=False)
-        self.wv = nn.Linear(args.dim, self.n_kv_heads * self.head_dim, bias=False)
-        self.wo = nn.Linear(args.n_heads * self.head_dim, args.dim, bias=False)
+        # self.dim = 4096, self.n_heads = 32, self.head_dim = 4096 / 32 = 125
```
Comment (helunwencser) [inline examples/models/llama/llama_transformer.py:274]: let's remove this comment since it is not true anymore. Different models have different configs for these values.

Comment (cccclai) [reply examples/models/llama/llama_transformer.py:274]: let me remove in the follow up diff
